src/solvers/flattening/boolbv_byte_extract.cpp

```diff
@@ -38,15 +38,15 @@ bvt map_bv(const bv_endianness_mapt &map, const bvt &src)
 
 bvt boolbvt::convert_byte_extract(const byte_extract_exprt &expr)
 {
+  const std::size_t width = boolbv_width(expr.type());
+
   // array logic does not handle byte operators, thus lower when operating on
   // unbounded arrays
-  if(is_unbounded_array(expr.op().type()))
+  if(is_unbounded_array(expr.op().type()) || is_unbounded_array(expr.type()) || width == 0)
```

Comment:
The test of `expr.type()` and `expr.op.type()` makes me a little nervous.  What about arrays at the start of structs or unions containing unions?

Reply:
I have expanded `is_unbounded_array` to cover these cases.